apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/PageQueryBuilder.java

```diff
@@ -0,0 +1,55 @@
+package gov.cms.bfd.server.war.stu3.providers;
+
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+public final class PageQueryBuilder<T> {
```

Comment:
Not sure what abstraction this class is trying to model. Is it trying to be a replacement for `CriteriaBuilder`. If we are going to make a builder, my suggestion is that some type of fluent class:  `builder().setValueA().setValueB().build()`

Reply:
The main thing it's _trying_ to hide is the handling of the three aspects of building these queries:
```
CriteriaBuilder
CriteriaQuery<T>
Root<T>
```
With an aim for readability and allowing things like `count` easily.

I like the fluent idea.